Architecture/TodoApp/android/core/src/main/java/com/novoda/todoapp/tasks/data/model/Tasks.java

```diff
@@ -121,12 +129,22 @@ public boolean apply(Map.Entry<Id, SyncedData<Task>> input) {
         };
     }
 
+    public Tasks overrideWithMostRecentActionsFrom(Tasks tasks, final long actionTimestamp) {
+        ImmutableMapWithCopy<Id, SyncedData<Task>> filter = tasks.internalMap().filter(new Predicate<Map.Entry<Id, SyncedData<Task>>>() {
```

Comment:
`filter` is a bad variable name here. How about `moreRecentActions` ? That way, line 139 will read `return Tasks.from(internalMap().putAll(moreRecentActions));` which makes it clearer that this `Tasks` object has been updated with the more recently updated `Task` objects from the parameter.